detection_rules/version_lock.py

```diff
@@ -97,6 +97,15 @@ def manage_versions(self, rules: RuleCollection,
                 lock_info = rule.contents.lock_info(bump=not exclude_version_update)
                 current_rule_lock: dict = current_version_lock.setdefault(rule.id, {})
 
+                # prevent rule type changes for already locked and released rules (#1854)
```

Comment:
The change in the `version_lock.py` assumes that no version changes are allowed across any stacks. Do we want to validate the min/max versions to support type changes for older stacks and new stacks (assuming it gets fixed in 8.2)?

Reply:
Base on the possible upgrade paths, I think it is better to just consistently prevent type changes altogether.